packages/expo-keep-awake/android/src/main/java/expo/modules/keepawake/ExpoKeepAwakeManager.java

```diff
@@ -0,0 +1,88 @@
+package expo.modules.keepawake;
+
+import android.app.Activity;
+import android.view.WindowManager;
+
+import org.unimodules.core.ModuleRegistry;
+import org.unimodules.core.Promise;
+import org.unimodules.core.errors.CurrentActivityNotFoundException;
+import org.unimodules.core.interfaces.ActivityProvider;
+import org.unimodules.core.interfaces.InternalModule;
+import org.unimodules.core.interfaces.ModuleRegistryConsumer;
+import org.unimodules.core.interfaces.services.KeepAwakeManager;
+
+import java.util.Collections;
+import java.util.List;
+
+public class ExpoKeepAwakeManager implements KeepAwakeManager, InternalModule, ModuleRegistryConsumer {
+
+  private final static String NO_ACTIVITY_ERROR_CODE = "NO_CURRENT_ACTIVITY";
+
+  private ModuleRegistry mModuleRegistry;
+  private boolean mIsActivated;
```

Comment:
How about making this a counter? 🤔

Reply:
I remember you suggested it. However, I am not sure if that's good idea. For several reasons:
1. iOS works with flag
2. We force certain behavior. Currently, developers can use this feature as they wish. They can implement any strategy for activiating/deactivating screen. Once we implement counter/latch/whatever it gets harder for them.

I think that if we wanted to make it 'smarter', we would need to at least add some forceActivate and forceDeactivate methods. Or add some tags mechanisms? Store given tag for activation and remove for deactivation, and actually deactivate when we removed the last tag?

Reply:
After reconsidering I think we should go with tags... Will do this.

Reply:
sounds good to me! 💯